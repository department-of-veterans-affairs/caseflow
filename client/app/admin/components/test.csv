VET_LAST_NAME,VET_FIRST_NAME,VET_MIDDLE_NAME,VET_DATE_OF_BIRTH,VET_SSN,VET_PARTICIPANT_ID,VET_FILE_NUMBER,APPELLANT_LAST_NAME,APPELLANT_FIRST_NAME,APPELLANT_MIDDLE_NAME,APPELLANT_DATE_OF_BIRTH,APPELLANT_SSN,APPELLANT_GENDER,APPELLANT_ADDRESS,APPELLANT_PHONE,APPELLANT_EMAIL,APPELLANT_EDI_PI,APPELLANT_CORP_PID,APPELLANT_VACOLS_INTERNAL_ID,RELATIONSHIP_TO_VETERAN