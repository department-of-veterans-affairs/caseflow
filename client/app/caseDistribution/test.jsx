/* eslint-disable react/prop-types */

import React from 'react';
import NavigationBar from '../components/NavigationBar';
import { BrowserRouter } from 'react-router-dom';
import PageRoute from '../components/PageRoute';
import AppFrame from '../components/AppFrame';
import AppSegment from '@department-of-veterans-affairs/caseflow-frontend-toolkit/components/AppSegment';
import { LOGO_COLORS } from '../constants/AppConstants';
import Footer from '@department-of-veterans-affairs/caseflow-frontend-toolkit/components/Footer';import CaseSearchLink from '../components/CaseSearchLink';

import CaseDistributionApp from './pages/CaseDistributionApp';

class CaseDistributionTest extends React.PureComponent {
  render() {
    const Router = this.props.router || BrowserRouter;
    const appName = 'Case Distribution';
    const tablestyle = {
      display: "block",
      overflowX: "scroll"
    };

    return (
      <Router {...this.props.routerTestProps}>
        <div>
          <NavigationBar
            wideApp
            defaultUrl={
              this.props.caseSearchHomePage || this.props.hasCaseDetailsRole ?
                '/search' :
                '/queue'
            }
            userDisplayName={this.props.userDisplayName}
            dropdownUrls={this.props.dropdownUrls}
            applicationUrls={this.props.applicationUrls}
            logoProps={{
              overlapColor: LOGO_COLORS.QUEUE.OVERLAP,
              accentColor: LOGO_COLORS.QUEUE.ACCENT,
            }}
            rightNavElement={<CaseSearchLink />}
            appName="Caseflow Admin"
          >
            <AppFrame>
              <AppSegment filledBackground>
                <div>
                  <PageRoute
                    exact
                    path={'/acd-controls/test'}
                    title="Case Distribution | Caseflow"
                    component={() => {
                      return (
                        <div>
                          <h1 id='top'>Case Distribution Levers Dashboard</h1>
                          <p>This page is for lower env use only and provides a convient place to review Case Distribution related data</p>
                          <h2>Navigation</h2>
                          <ul>
                            <li>
                              <a href="#levers"><button className="btn btn-primary">Case Distribution Levers</button></a>
                            </li>
                            <li>
                              <a href="#audit"><button className="btn btn-primary">Case Distribution Audit Levers</button></a>
                            </li>
                            <li>
                             <a href="#distribution_status"><button className="btn btn-primary">Distribution Status</button></a>
                            </li>
                          </ul>
                          <h2 id='levers'> Case Distribution Levers </h2>
                            <table id="case-table-description" className="usa-table-borderless undefined" style={tablestyle}>
                              <thead>
                                <tr>
                                  <td> ID </td>
                                  <td> Title </td>
                                  <td> Item </td>
                                  <td> Control Group </td>
                                  <td> Lever Group </td>
                                  <td> Lever Group Order </td>
                                  <td> Description </td>
                                  <td> Data Type </td>
                                  <td> Value </td>
                                  <td> Min / Max Value </td>
                                  <td> Unit </td>
                                  <td> Option Values</td>
                                  <td> Toggle Active </td>
                                  <td> Disabled In UI</td>
                                  <td> Algorithms Used </td>
                                  <td> Created At </td>
                                  <td> Updated At </td>
                                </tr>
                              </thead>
                              <tbody>
                              {this.props.acdLevers.map((lever, i) => {
                                return [
                                  <tr>
                                    <td> { lever.id } </td>
                                    <td> { lever.title } </td>
                                    <td> { lever.item } </td>
                                    <td> { lever.control_group} </td>
                                    <td> { lever.lever_group } </td>
                                    <td> { lever.lever_group_order } </td>
                                    <td> { lever.description } </td>
                                    <td> { lever.data_type } </td>
                                    <td> { lever.value } </td>
                                    <td> { lever.min_value }/{lever.max_value } </td>
                                    <td> { lever.unit } </td>
                                    <td> { lever.options?.map((option) => option.value).join(', ') } </td>
                                    <td> { lever.is_toggle_active.toString() } </td>
                                    <td> { lever.is_disabled_in_ui.toString()  } </td>
                                    <td> { lever.algorithms_used } </td>
                                    <td> { lever.created_at } </td>
                                    <td> { lever.updated_at } </td>
                                  </tr>
                                ];
                              })}
                              </tbody>
                            </table>
                          <hr/>
                          <h2 id='audit'> Case Distribution Audit Levers </h2>
                          <table>
                            <thead>
                              <tr>
                                <td> ID </td>
                                <td> Lever ID </td>
                                <td> Created At </td>
                                <td> Previous Value</td>
                                <td> Update Value </td>
                                <td> User CSS ID </td>
                                <td> User Name</td>
                                <td> Lever Title </td>
                                <td> Lever Data Type </td>
                                <td> Lever Unit </td>
                              </tr>
                            </thead>
                            <tbody>
                            {this.props.acdHistory.map((entry, i) => {
                              return [
                                  <tr>
                                    <td> { entry.id } </td>
                                    <td> { entry.case_distribution_lever_id } </td>
                                    <td> { entry.created_at } </td>
                                    <td> { entry.previous_value } </td>
                                    <td> { entry.update_value } </td>
                                    <td> { entry.user_css_id } </td>
                                    <td> { entry.user_name } </td>
                                    <td> { entry.lever_title } </td>
                                    <td> { entry.lever_data_type } </td>
                                    <td> { entry.lever_unit } </td>
                                  </tr>
                              ];
                            })}

                            </tbody>
                          </table>
                          <hr/>
                          <h2 id='distribution_status'>Distribution Status</h2>
                          <ul>
                            <li>
                              <a href="/appeals-ready-to-distribute?csv=1"><button className="btn btn-primary">Download Appeals Ready to Distribute CSV</button></a>
                            </li>
                            <li>
                              <a href="/appeals-distributed?csv=1"><button className="btn btn-primary">Download Distributed Appeals CSV</button></a>
                            </li>
                          </ul>
                          <hr/>
                          <a href="#top"><button className="btn btn-primary">Back to Top</button></a>
                        </div>
                      );
                    }}
                  />
                </div>
              </AppSegment>
            </AppFrame>
          </NavigationBar>
          <Footer
            appName={appName}
            feedbackUrl={this.props.feedbackUrl}
            buildDate={this.props.buildDate}
          />
        </div>
      </Router>

    );

  }
}

export default CaseDistributionTest;
