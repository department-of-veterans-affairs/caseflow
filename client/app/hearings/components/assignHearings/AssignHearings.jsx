import { css } from 'glamor';
import PropTypes from 'prop-types';
import React from 'react';
import _ from 'lodash';
import moment from 'moment';

import { COLORS } from '../../../constants/AppConstants';
import { NoUpcomingHearingDayMessage } from './Messages';
import AssignHearingsTabs from './AssignHearingsTabs';
import Button from '../../../components/Button';

const sectionNavigationListStyling = css({
  '& > li': {
    backgroundColor: COLORS.GREY_BACKGROUND,
    color: COLORS.PRIMARY,
    borderWidth: 0
  }
});

const buttonColorSelected = css({
  backgroundColor: COLORS.GREY_DARK,
  color: COLORS.WHITE,
  borderRadius: '0.1rem 0.1rem 0 0',
  '&:hover': {
    backgroundColor: COLORS.GREY_DARK,
    color: COLORS.WHITE
  }
});

const roSelectionStyling = css({ marginTop: '10px' });

const UpcomingHearingDaysNav = ({
  upcomingHearingDays, selectedHearingDay,
  onSelectedHearingDayChange
}) => (
  <div className="usa-width-one-fourth" {...roSelectionStyling}>
    <h3>Hearings to Schedule</h3>
    <h4>Available Hearing Days</h4>
    <ul className="usa-sidenav-list" {...sectionNavigationListStyling}>
      {_.orderBy(Object.values(upcomingHearingDays), (hearingDay) => hearingDay.scheduledFor, 'asc').
        map((hearingDay) => {
          const dateSelected = selectedHearingDay &&
          (selectedHearingDay.scheduledFor === hearingDay.scheduledFor &&
             selectedHearingDay.room === hearingDay.room);

          return <li key={hearingDay.id} >
            <Button
              styling={dateSelected ? buttonColorSelected : {}}
              onClick={() => onSelectedHearingDayChange(hearingDay)}
              linkStyling>
              {`${moment(hearingDay.scheduledFor).format('ddd M/DD/YYYY')}
              ${hearingDay.room}`}
            </Button>
          </li>;
        })}
    </ul>
  </div>
);

UpcomingHearingDaysNav.propTypes = {
  upcomingHearingDays: PropTypes.object,
  selectedHearingDay: PropTypes.shape({
    scheduledFor: PropTypes.string,
    room: PropTypes.string
  }),
  onSelectedHearingDayChange: PropTypes.func
};

export default class AssignHearings extends React.Component {

  room = () => {
    const { selectedRegionalOffice, selectedHearingDay } = this.props;

    // St. Petersburg, FL or Winston-Salem, NC
    if (selectedRegionalOffice === 'RO17' || selectedRegionalOffice === 'RO18') {
      if (selectedHearingDay) {
        return selectedHearingDay.room;
      }
    }

    return '';
  };

  render() {
    const {
      upcomingHearingDays,
      selectedHearingDay,
      selectedRegionalOffice,
      onSelectedHearingDayChange
    } = this.props;
    const room = this.room();

    if (_.isEmpty(upcomingHearingDays)) {
      return <NoUpcomingHearingDayMessage />;
    }

    return (
      <React.Fragment>
        <UpcomingHearingDaysNav
          upcomingHearingDays={upcomingHearingDays}
          selectedHearingDay={selectedHearingDay}
          onSelectedHearingDayChange={onSelectedHearingDayChange} />
        <AssignHearingsTabs
          selectedRegionalOffice={selectedRegionalOffice}
          selectedHearingDay={selectedHearingDay}
          room={room}
        />
      </React.Fragment>
    );
  }
}

AssignHearings.propTypes = {
  selectedRegionalOffice: PropTypes.string,
  upcomingHearingDays: PropTypes.object,
  onSelectedHearingDayChange: PropTypes.func,
  selectedHearingDay: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object
  ]),
  userId: PropTypes.number
};
