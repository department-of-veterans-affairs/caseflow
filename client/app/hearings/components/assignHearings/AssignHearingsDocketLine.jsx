import { css } from 'glamor';
import _ from 'lodash';
import moment from 'moment';

import LEGACY_APPEAL_TYPES_BY_ID from '../../../../constants/LEGACY_APPEAL_TYPES_BY_ID';

export const docketCutoffLineStyle = (_index, scheduledForText) => {
  const isEmpty = _index < 0;
  const index = isEmpty ? 0 : _index;
  const style = {
    [`& #table-row-${index + 1} td`]: {
      paddingTop: '35px'
    },
    [`& #table-row-${index}`]: {
      borderBottom: '2px solid #000',
      position: 'relative',
      '& td': {
        paddingBottom: '35px'
      },
      '& td:first-child::before': {
        content: `Schedule for ${scheduledForText}`,
        display: 'block',
        position: 'absolute',
        transform: 'translateY(calc(100% + 4px))',
        background: '#fff',
        padding: '10px 10px 10px 0px',
        height: '42px',
        fontWeight: 'bold'
      }
    }
  };

  const isEmptyStyle = isEmpty ? {
    '& th': {
      paddingBottom: '35px'
    },
    [`& #table-row-${index}`]: {
      borderTop: '2px solid #000',
      borderBottom: 0,
      '& td': {
        paddingTop: '35px',
        paddingBottom: '15px'
      },
      '& td:first-child::before': {
        transform: 'translateY(-165%)',
        content: `All veterans have been scheduled through ${scheduledForText}`
      }
    }
  } : {};

  return css(_.merge(style, isEmptyStyle));
};

const getAppealsInDocketRange = (appeals, endOfRange) => _.filter(appeals, (appeal) => (
  moment(appeal.docketRangeDate).isBefore(endOfRange) ||
  appeal.caseType === LEGACY_APPEAL_TYPES_BY_ID.cavc_remand ||
  appeal.aod
));

export const getIndexOfDocketLine = (appeals, endOfRange) => {
  if (_.isEmpty(appeals)) {
    return -1;
  }

  const numberOfAppealsInDocketRange = getAppealsInDocketRange(appeals, endOfRange).length;

  return numberOfAppealsInDocketRange - 1;
};
