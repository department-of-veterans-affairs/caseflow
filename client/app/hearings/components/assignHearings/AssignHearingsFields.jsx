import PropTypes from 'prop-types';
import React from 'react';
import _ from 'lodash';

import DocketTypeBadge from '../../../components/DocketTypeBadge';

export const HearingAppellantName = ({ hearing }) => {
  let { appellantFirstName, appellantLastName, veteranFirstName, veteranLastName, veteranFileNumber } = hearing;

  let appellantName;

  if (appellantFirstName && appellantLastName) {
    appellantName = `${appellantFirstName} ${appellantLastName} | ${veteranFileNumber}`;
  } else if (veteranFirstName && veteranLastName) {
    appellantName = `${veteranFirstName} ${veteranLastName} | ${veteranFileNumber}`;
  } else {
    appellantName = veteranFileNumber;
  }

  return <React.Fragment>{appellantName}</React.Fragment>;
};

HearingAppellantName.propTypes = {
  hearing: PropTypes.shape({
    appellantFirstName: PropTypes.string,
    appellantLastName: PropTypes.string,
    veteranFirstName: PropTypes.string,
    veteranLastName: PropTypes.string,
    veteranFileNumber: PropTypes.string
  })
};

export const CaseDetailsInformation = ({ appeal }) => {
  let caseDetails;

  if (appeal.appellantFullName) {
    caseDetails = `${appeal.appellantFullName} | ${appeal.veteranFileNumber}`;
  } else {
    caseDetails = `${appeal.veteranFullName} | ${appeal.veteranFileNumber}`;
  }

  return <React.Fragment>{caseDetails}</React.Fragment>;
};

CaseDetailsInformation.propTypes = {
  appeal: PropTypes.shape({
    appellantFullName: PropTypes.string,
    veteranFullName: PropTypes.string,
    veteranFileNumber: PropTypes.string
  })
};

export const HearingDocketTag = ({ hearing }) => {
  if (hearing.docketNumber) {
    return <div>
      <DocketTypeBadge name={hearing.docketName} number={hearing.docketNumber} />
      {hearing.docketNumber}
    </div>;
  }

  return null;
};

HearingDocketTag.propTypes = {
  hearing: PropTypes.shape({
    docketName: PropTypes.string,
    docketNumber: PropTypes.oneOfType([
      PropTypes.number,
      PropTypes.string
    ])
  })
};

export const AppealDocketTag = ({ appeal }) => {
  if (appeal.docketNumber) {
    return <div>
      <DocketTypeBadge name={appeal.docketName} number={appeal.docketNumber} />
      {appeal.docketNumber}
    </div>;
  }

  return null;
};

AppealDocketTag.propTypes = {
  appeal: PropTypes.shape({
    docketName: PropTypes.string,
    docketNumber: PropTypes.string
  })
};

export const SuggestedHearingLocation = ({ suggestedLocation, format }) => {

  if (!suggestedLocation) {
    return null;
  }

  return (
    <React.Fragment>
      <div>{`${format(suggestedLocation)}`}</div>
      {!_.isNil(suggestedLocation.distance) &&
        <div>{`Distance: ${suggestedLocation.distance} miles away`}</div>
      }
    </React.Fragment>
  );
};

SuggestedHearingLocation.propTypes = {
  format: PropTypes.func.isRequired,
  suggestedLocation: PropTypes.shape({
    name: PropTypes.string,
    address: PropTypes.string,
    city: PropTypes.string,
    state: PropTypes.string,
    zipCode: PropTypes.string,
    distance: PropTypes.number,
    classification: PropTypes.string,
    facilityId: PropTypes.string,
    facilityType: PropTypes.string
  })
};
