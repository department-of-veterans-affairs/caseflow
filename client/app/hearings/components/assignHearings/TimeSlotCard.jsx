import React from 'react';
import PropTypes from 'prop-types';

import { TimeSlotDetail } from '../scheduleHearing/TimeSlotDetail';
import { HearingAppellantName } from './AssignHearingsFields';
import { HearingTime } from '../HearingTime';
import { Dot } from '../../../components/Dot';

export const TimeSlotCard = ({ hearing, hearingDay, regionalOffice }) => {
  return (
    <div className="usa-grid time-slot-card">
      <div className="usa-width-one-fourth">
        <HearingTime
          primaryLabel="RO"
          hearing={hearing}
          paragraphClasses="time-slot-card-time"
          labelClasses="time-slot-card-label"
          breakCharacter=""
        />
      </div>
      <div className="usa-width-three-fourths">
        <TimeSlotDetail
          {...hearing}
          hearingDay={hearingDay}
          regionalOffice={regionalOffice}
          issueCount={hearing.currentIssueCount}
          showType
          showDetails
          itemSpacing={5}
          hearing={hearing}
          label={
            <span className="time-slot-card-label">
              <HearingAppellantName
                hearing={{
                  ...hearing,
                  veteranFileNumber: `Veteran ID: ${hearing.veteranFileNumber}`,
                }}
                spacingCharacter={<Dot spacing={2} />}
              />
            </span>
          }
        />
      </div>
    </div>
  );
};

TimeSlotCard.propTypes = {
  hearing: PropTypes.object,
  hearingDay: PropTypes.object,
  regionalOffice: PropTypes.string,
};
