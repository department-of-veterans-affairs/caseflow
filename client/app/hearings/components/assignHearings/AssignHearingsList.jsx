// External Dependencies
import React from 'react';
import PropTypes from 'prop-types';
import { sortBy } from 'lodash';

// Component Dependencies
import { TimeSlotCard } from './TimeSlotCard';

/**
 * Assign Hearings List Component
 * @param {Object} props -- Contains the hearings list, selected hearing day and regional office
 */
export const AssignHearingsList = ({ hearings,
  hearingDay,
  regionalOffice,
  mstIdentification,
  pactIdentification,
  legacyMstPactIdentification }) => {
  return hearings.length ? sortBy(hearings, 'scheduledTimeString').map((hearing) => (
    <TimeSlotCard
      key={hearing.appealExternalId}
      hearing={hearing}
      hearingDay={hearingDay}
      regionalOffice={regionalOffice}
      mstIdentification={mstIdentification}
      pactIdentification={pactIdentification}
      legacyMstPactIdentification={legacyMstPactIdentification}
    />
  )) : (
    <div className="time-slot-card-label no-hearings-label">No Upcoming hearings to display</div>
  );
};

AssignHearingsList.propTypes = {
  hearings: PropTypes.array,
  hearingDay: PropTypes.object,
  regionalOffice: PropTypes.string,
  mstIdentification: PropTypes.bool,
  pactIdentification: PropTypes.bool,
  legacyMstPactIdentification: PropTypes.bool
};
