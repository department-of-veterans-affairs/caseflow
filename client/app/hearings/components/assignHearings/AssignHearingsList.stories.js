import React from 'react';

import { AssignHearingsList } from './AssignHearingsList';
import { amaHearing, legacyHearing, defaultHearing } from '../../../../test/data/hearings';

export default {
  title: 'Hearings/Components/Assign Hearings/AssignHearingsList',
  component: AssignHearingsList,
  argTypes: {
    appeal: { table: { disable: true } },
  }
};

const Template = (args) => {
  return (
    <AssignHearingsList
      {...args}
    />
  );
};

export const Basic = Template.bind({});
Basic.args = {
  hearings: [defaultHearing, amaHearing, legacyHearing]
};
Basic.argTypes = {
};
