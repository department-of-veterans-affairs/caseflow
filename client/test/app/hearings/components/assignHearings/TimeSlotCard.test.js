import React from 'react';

import { TimeSlotCard } from 'app/hearings/components/assignHearings/TimeSlotCard';
import { render } from '@testing-library/react';
import { getHearingDetailsArray, getHearingType, defaultHearing, defaultHearingDay } from 'test/data';

// St. Petersburg FL
const regionalOffice = 'RO17';

describe('TimeSlotCard', () => {
  test('Matches snapshot with default props', () => {
    // Run the test
    const card = render(
      <TimeSlotCard regionalOffice={regionalOffice} hearing={defaultHearing} hearingDay={defaultHearingDay} />
    );

    // Row 1 data
    const veteranName = `${defaultHearing.appellantFirstName} ${defaultHearing.appellantLastName}`;
    const headerLabel = `${veteranName} · Veteran ID: ${defaultHearing.veteranFileNumber}`;

    // Row 2 data
    const secondRowLabelArray = getHearingDetailsArray(defaultHearing, true);

    // Row 3 data
    const lastRowLabel = getHearingType(defaultHearing);

    expect(document.getElementsByClassName('usa-width-one-fourth')).toHaveLength(1);
    expect(document.getElementsByClassName('usa-width-three-fourths')).toHaveLength(1);
    expect(document.getElementsByClassName('time-slot-card-time')).toHaveLength(1);
    expect(document.getElementsByClassName('time-slot-card-label')).toHaveLength(2);
    expect(card.getAllByText('·')).toHaveLength(5);

    // Row 1
    expect(document.getElementsByClassName('time-slot-card-label')[1]).toHaveTextContent(headerLabel);

    // Row 2
    secondRowLabelArray.forEach((elem) => {
      expect(document.getElementsByClassName('time-slot-details')[0]).toHaveTextContent(elem);
    });

    // Row 3
    expect(document.getElementsByClassName('time-slot-details')[1]).toHaveTextContent(lastRowLabel);

    expect(card).toMatchSnapshot();
  });
});
