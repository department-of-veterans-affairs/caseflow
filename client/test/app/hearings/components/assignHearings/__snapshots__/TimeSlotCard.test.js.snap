// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeSlotCard Matches snapshot with default props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="usa-grid time-slot-card"
      >
        <div
          class="usa-width-one-fourth"
        >
          <div>
            <p
              class="time-slot-card-time"
            >
              <span
                class="time-slot-card-label"
              >
                6:00 AM EDT
              </span>
            </p>
          </div>
        </div>
        <div
          class="usa-width-three-fourths"
        >
          <span
            class="time-slot-card-label"
          >
            Tom Brady
             
            <span
              style="margin: 0px 2px;"
            >
              ·
            </span>
             
            Veteran ID: 100000005
          </span>
          <div
            class="time-slot-details"
          >
            0 issues
             
            <span
              style="margin: 0px 5px;"
            >
              ·
            </span>
             
            <span
              aria-describedby="badge-200624-614"
              currentitem="false"
              data-event="focus mouseenter"
              data-event-off="keydown mouseleave"
              data-for="badge-200624-614"
              data-tip="true"
              role="tooltip"
              tabindex="0"
            >
              <span
                aria-label="Hearing Request"
                data-css-2moai0=""
              >
                H
              </span>
            </span>
            <span
              data-css-1ediyry=""
            >
              <div
                class="__react_component_tooltip place-bottom type-dark "
                data-id="tooltip"
                id="badge-200624-614"
              >
                Hearing Request
              </div>
            </span>
             
            200624-614
             
            <span
              style="margin: 0px 5px;"
            >
              ·
            </span>
             
            AMERICAN LEGION
          </div>
          <div
            class="time-slot-details"
          >
            <span>
              Original
            </span>
             
            <span
              style="margin: 0px 5px;"
            >
              ·
            </span>
             
            <span>
              Video
            </span>
             
            <span
              style="margin: 0px 5px;"
            >
              ·
            </span>
             
            <a
              href="/queue/appeals/0bf0263c-d863-4405-9b2e-f55cff77c6c4/?hearingDate=2020-08-15&regionalOffice=RO17"
              id="0bf0263c-d863-4405-9b2e-f55cff77c6c4"
            >
              View Case Details
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="usa-grid time-slot-card"
    >
      <div
        class="usa-width-one-fourth"
      >
        <div>
          <p
            class="time-slot-card-time"
          >
            <span
              class="time-slot-card-label"
            >
              6:00 AM EDT
            </span>
          </p>
        </div>
      </div>
      <div
        class="usa-width-three-fourths"
      >
        <span
          class="time-slot-card-label"
        >
          Tom Brady
           
          <span
            style="margin: 0px 2px;"
          >
            ·
          </span>
           
          Veteran ID: 100000005
        </span>
        <div
          class="time-slot-details"
        >
          0 issues
           
          <span
            style="margin: 0px 5px;"
          >
            ·
          </span>
           
          <span
            aria-describedby="badge-200624-614"
            currentitem="false"
            data-event="focus mouseenter"
            data-event-off="keydown mouseleave"
            data-for="badge-200624-614"
            data-tip="true"
            role="tooltip"
            tabindex="0"
          >
            <span
              aria-label="Hearing Request"
              data-css-2moai0=""
            >
              H
            </span>
          </span>
          <span
            data-css-1ediyry=""
          >
            <div
              class="__react_component_tooltip place-bottom type-dark "
              data-id="tooltip"
              id="badge-200624-614"
            >
              Hearing Request
            </div>
          </span>
           
          200624-614
           
          <span
            style="margin: 0px 5px;"
          >
            ·
          </span>
           
          AMERICAN LEGION
        </div>
        <div
          class="time-slot-details"
        >
          <span>
            Original
          </span>
           
          <span
            style="margin: 0px 5px;"
          >
            ·
          </span>
           
          <span>
            Video
          </span>
           
          <span
            style="margin: 0px 5px;"
          >
            ·
          </span>
           
          <a
            href="/queue/appeals/0bf0263c-d863-4405-9b2e-f55cff77c6c4/?hearingDate=2020-08-15&regionalOffice=RO17"
            id="0bf0263c-d863-4405-9b2e-f55cff77c6c4"
          >
            View Case Details
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
